rfcs/text/0006_management_section_service.md

```diff
@@ -0,0 +1,341 @@
+
+## Why now?
+The main driver for considering this now is that the Management API moving to the new platform is going to block other teams from completing migration, so we need to have an answer to what the new platform version of the API looks like as soon as possible in `7.x`.
+
+## Why not just keep the current API and redesign later?
+The answer to that has to do with the items that are currently used in the management implementation which must be removed in order to migrate to NP: the framework currently registers a `uiExport`, and relies on `IndexedArray`, `uiRegistry`, and `ui/routes`.
+
+This means that we will basically need to rebuild the service anyway in order to migrate to the new platform. So if we are going to invest that time, we might as well invest it in building the API the way we want it to be longer term, rather than creating more work for ourselves later.
+
+## Technical goals
+- Remove another usage of `IndexedArray` & `uiRegistry` (required for migration)
+- Remove dependency on `ui/routes` (required for migration)
+- Remove management section `uiExport` (required for migration)
+- Simple API that is designed in keeping with new platform principles
+  - This includes being rendering-framework-agnostic... You should be able to build your management section UI however you'd like
+- Clear separation of app/UI code and service code, even if both live within the same plugin
+- Flexibility to potentially support alternate layouts in the future (see mockups in [reference section](#reference) below)
+
+# Basic example
+This API is influenced heavily by the [application service mounting RFC](https://github.com/elastic/kibana/blob/master/rfcs/text/0004_application_service_mounting.md). The intent is to make the experience consistent with that service; the Management section is basically one big app with a bunch of registered "subapps".
+
+```ts
+// my_plugin/public/plugin.ts
+
+export class MyPlugin {
+  setup(core, { management }) {
+    // Registering a new link to a new section
+    const mySection = management.sections.register({
+      id: 'my-section',
+      title: 'My Main Section', // display name
+      description: 'hello', // not used in current UI, but possibly in future
+      order: 10,
+      euiIconType: 'iconName',
+    });
+    mySection.registerLink({
+      id: 'my-link',
+      title: 'My Link', // display name
+      description: 'hello', // not used in current UI, but possibly in future
+      order: 20,
+      async mount(context, params) {
+        const { renderApp } = await import('./my-section');
+        return renderApp(context, params);
+      }
+    });
+
+    // Registering a new link to an existing section
+    const kibanaSection = management.sections.get('kibana');
+    kibanaSection.registerLink({ id: 'my-kibana-section-link', ... });
+  }
+
+  start(core, { management }) {
+    // access all registered sections, filtered based on capabilities
+    management.sections.getAvailable();
```

Comment:
I'm not entirely sure how to interpret this example. In a real-world scenario would we assign the value returned by `getAvailable` to a variable and act upon it?

Are there examples in the codebase where people have to deal with the problem this method solves? Can we update the "Legacy service" section example with some demonstrating how we currently solve this problem?

Reply:
lets add a better example, this should retrieve all sections, which management app then loops over and shows them in ui.

Reply:
resolved